qiskit/passmanager/passmanager.py

```diff
@@ -212,7 +212,7 @@ def callback_func(**kwargs):
             return in_programs
 
         is_list = True
-        if not isinstance(in_programs, Sequence):
+        if not (isinstance(in_programs, Sequence) and not isinstance(in_programs, str)):
```

Comment:
A classic problem with trying to accept `T | Sequence[T]`, haha; the type is ambiguous if `T: Sequence`.  There's other types that this would hit; if somebody wants to use an IR type that's a `list`, say, it'd hit the same problem.

It's worth considering whether a more complete solution would be to tighten the typing to `T | list[T]`, and say "`list` is not a valid IR type".

Reply:
That sounds like a better solution. Done in [6838e55](https://github.com/Qiskit/qiskit/pull/11178/commits/6838e55a864626b423e7f9d9131fc5416f4d81c2)